### ISSUE TITLE
[pytorch 0.4.0] Cuda runtime error (30) after upgrading from 0.3.1

### ISSUE THREAD

**Issue by gmkim90** [pytorch/pytorch#7031]
## Issue description
I am using python 3.6.5 on Ubuntu 14.04.
I upgraded pytorch 0.4.0 via ‘pip uninstall torch torchvision; pip install torch torchvision’

After upgrade, cuda operation for tensor or model gives me cuda runtime error(30).

When I downgrade to 0.3.0.post4, tensor.cuda() makes no error.

## Code example
device = torch.device(‘cuda:0’)
aaa = torch.Tensor(5).to(device)

Traceback (most recent call last):
File “”, line 1, in 
File “/home/kenkim/anaconda3/lib/python3.6/site-packages/torch/cuda/init.py”, line 161, in _lazy_init
torch._C._cuda_init()
RuntimeError: cuda runtime error (30) : unknown error at /pytorch/aten/src/THC/THCGeneral.cpp:844

## System Info
PyTorch version: 0.4.0
Is debug build: No
CUDA used to build PyTorch: 8.0.61

OS: Ubuntu 14.04.5 LTS
GCC version: (GCC) 4.8.5
CMake version: version 2.8.12.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: TITAN X (Pascal)
GPU 1: TITAN X (Pascal)

Nvidia driver version: 384.111
cuDNN version: Probably one of the following:
/usr/local/MATLAB/R2017a/bin/glnxa64/libcudnn.so.5.1.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.0.5
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn.so.5.1.10
/usr/local/cuda-8.0/targets/x86_64-linux/lib/libcudnn_static.a
/usr/local/cudnn/lib64/libcudnn.so.5.1.10
/usr/local/cudnn/lib64/libcudnn_static.a

Versions of relevant libraries:
[pip3] numpy (1.13.3)
[pip3] numpydoc (0.6.0)
[pip3] torch (0.4.0)
[pip3] torchaudio (0.1)
[pip3] torchfile (0.1.0)
[pip3] torchvision (0.2.1)
[pip3] warpctc-pytorch (0.1)
[conda] torch                     0.4.0                     <pip>
[conda] torchaudio                0.1                       <pip>
[conda] torchfile                 0.1.0                     <pip>
[conda] torchvision               0.2.1                     <pip>
[conda] warpctc_pytorch           0.1                       <pip>

**Comment by fmassa** [comment]
I'm not sure if this is expected, but it looks like you have installed torch (and torchvision etc) twice using both `pip` and `conda`. Could you try removing them all and installing again?

**Comment by gmkim90** [comment]
After removing conda, pip pytorch (conda uninstall pytorch; pip uninstall torch torchvision), install pytorch 0.3.1 (via pip) & execute python collect_env.py, result still show torch, torchvision in both pip3 & conda.

Versions of relevant libraries:
[pip3] numpy (1.13.3)
[pip3] numpydoc (0.6.0)
[pip3] torch (0.3.1)
[pip3] torchaudio (0.1)
[pip3] torchfile (0.1.0)
[pip3] torchvision (0.2.1)
[pip3] warpctc-pytorch (0.1)
[conda] torch                     0.3.1                     <pip>
[conda] torchaudio                0.1                       <pip>
[conda] torchfile                 0.1.0                     <pip>
[conda] torchvision               0.2.1                     <pip>
[conda] warpctc_pytorch           0.1                       <pip>

**Comment by soumith** [comment]
this is indeed strange. Cuda runtime error 30 is most frustrating because it is Unknown Error.
Is there any chance that you can upgrade your CUDA driver and see if that helps?

**Comment by zou3519** [comment]
@fmassa if you do a conda install of pytorch/torchvision, it'll show up in both your `conda list` and `pip list` which is what's happening here

**Comment by xulu0620** [comment]
@gmkim90 I came across the same problem after upgrading from 0.3.1 to 0.4. Do you solve this problem now ?

**Comment by gmkim90** [comment]
@lucasxlu Sorry I couldn't work on this issue for a while.

**Comment by bryancresswell** [comment]
Facing the same issue. Please advise.

**Comment by matt-erhart** [comment]
Same here. Windows 10. Cuda 9.0. fresh install today.

**Comment by eli5168** [comment]
Facing the same issue.  QAQ

**Comment by matt-erhart** [comment]
Maybe unrelated but I needed this on windows:
torch.backends.cudnn.enabled = False # deal with cudnn X pytorch .4 bug. Might work with pytorch update.

**Comment by thongnnguyen** [comment]
My situation was similar. But my starting point may be different than everyone here so I was able to receover `0.4.0` working with CUDA. Just want to tell it for reference. I was having pytorch 0.4.0 working well with cuda (since its release) (I didn't use GPU when I was using 0.3.1. Only when I installed 0.4.0 I decided to invest a GPU)

Curerent, from `nvcc --version`:
```
Cuda compilation tools, release 9.2, V9.2.88
```
From `nvidia-smi`:
```
NVIDIA-SMI 390.67                 Driver Version: 390.67                    
```
Then a week ago, I was force to reinstall conda by another incident. Then with fresh installed conda, I install pytorch (Sep 2018) by (from [pytorch.org](pytorch.org))
```
conda install pytorch torchvision cuda92 -c pytorch
```
It installs pytorch `0.4.1py36_cuda9.2.148_cudnn7.1.4_1` and I was hit with the same error everyone here has.

I then install remove it and install `0.4.0` by
```
conda install pytorch=0.4.0
```
It installs `0.4.0` and my old code works on GPU again.

So, whatever it is, it may not happen only going from `0.3.1` to `0.4.0` but also going from `0.4.0` to `0.4.1`.
